source/fuzz/fuzzer_pass_donate_modules.cpp

```diff
@@ -413,14 +414,43 @@ void FuzzerPassDonateModules::HandleTypeOrValue(
             argument_type_ids));
       }
     } break;
+    case SpvOpSpecConstantOp: {
+      new_result_id = GetFuzzerContext()->GetFreshId();
+      auto type_id = original_id_to_donated_id->at(type_or_value.type_id());
+      auto opcode = static_cast<SpvOp>(type_or_value.GetSingleWordInOperand(0));
+
+      // Make sure we take into account |original_id_to_donated_id| when
+      // computing operands for OpSpecConstantOp.
+      opt::Instruction::OperandList operands;
+      for (uint32_t i = 1; i < type_or_value.NumInOperands(); ++i) {
+        const auto& operand = type_or_value.GetInOperand(i);
+        opt::Operand::OperandData data;
+
+        if (operand.type == SPV_OPERAND_TYPE_ID) {
+          data = {original_id_to_donated_id->at(operand.words[0])};
+        } else {
+          data = operand.words;
+        }
```

Comment:
I would do this as a ternary:

auto data = operand.type == SPV_OPERAND_TYPE_ID ? ... : ...;

Up to you, though.

Reply:
Fixed in d4ad88f.